### PR TITLE
docs: add documentation on developing Talos

### PR DESCRIPTION
**Pull request by smira**
Target audidence:

* maintainers
* contributors

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4826)

### CODE REVIEW

File: website/content/docs/v0.14/Learn More/developing-talos.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+```bash
+docker run -d -p 5005:5005 \
+    --restart always \
+    --name local registry:2
+```
+
+Try to build and push to local registry an installer image:
+
+```bash
+make installer IMAGE_REGISTRY=127.0.0.1:5005 PUSH=true
```
Comment (Unix4ever) [inline website/content/docs/v0.14/Learn More/developing-talos.md:45]: I think it also worths mentioning that you can pin the tag.
Not a big deal, but it's easier sometimes if you don't want to copy/paste the full image path everytime.

File: website/content/docs/v0.14/Learn More/developing-talos.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+`make <something> WITH_RACE=1` enables Go race detector, Talos runs slower and uses more memory, but memory races are detected.
+
+`make <something> WITH_DEBUG=1` enables Go profiling and other debug features, useful for local development.
+
+## Destroying Custer
+
+```bash
+sudo -E ../talos/_out/talosctl-linux-amd64 cluster destroy --provisioner=qemu
+```
```
Comment (Unix4ever) [inline website/content/docs/v0.14/Learn More/developing-talos.md:181]: worth mentioning that if you rebooted the machine, you need to manualy cleanup `~/.talos/clusters/<name>` dir

**Review by Unix4ever — APPROVED** [review]

File: website/content/docs/v0.14/Learn More/developing-talos.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+Some steps might work under Mac OS X, but using Linux is highly advised.
+
+## Prepare
+
+Check out the [Talos repository](https://github.com/talos-systems/talos).
+
+Try running `make help` to see available `make` commands.
+You need Docker and `buildx` installed on the host.
+
+> Note: On Ubuntu 21.10 it's easier to use Ubuntu's `docker`, for earlier versions of Ubuntu use [Docker's apt repositories](https://docs.docker.com/engine/install/ubuntu/).
```
Comment (frezbo) [inline website/content/docs/v0.14/Learn More/developing-talos.md:17]: let's remove this

Comment (smira) [reply website/content/docs/v0.14/Learn More/developing-talos.md:17]: I updated that to use different wording, but I think it still makes sense to use Docker from upstream.

File: website/content/docs/v0.14/Learn More/developing-talos.md (hunk 2 above)
Comment (smira) [reply website/content/docs/v0.14/Learn More/developing-talos.md:181]: thanks, I'll add that

**Review by frezbo — APPROVED** [review]

**Pull request merged**